firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -219,6 +220,16 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 browserViewType: state.browserViewType,
                 navigateTo: .home,
                 microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action))
+        case GeneralBrowserActionType.addNewTab:
+            return BrowserViewControllerState(
+                searchScreenState: state.searchScreenState,
+                showDataClearanceFlow: state.showDataClearanceFlow,
+                fakespotState: state.fakespotState,
+                toast: state.toast,
+                windowUUID: state.windowUUID,
+                browserViewType: state.browserViewType,
+                navigateTo: .newTab,
+                microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action))
```

Comment:
nit: maybe add a test here that calling `addNewTab` causes the state to be `.newTab`?

Reply:
thanks for adding tests!